packages/eslint-config-universe/__tests__/tools/checkPrettierRulesAsync.js

```diff
@@ -7,17 +7,7 @@ module.exports = async function checkPrettierRulesAsync(configFile) {
     PATH: path.resolve(__dirname, '../../node_modules/.bin') + path.delimiter + process.env.PATH,
   });
 
-  const { stdout: configString } = await spawnAsync(
-    'eslint',
-    ['--config', configFile, '--no-eslintrc', '--print-config', configFile],
-    { env }
-  );
-
-  const resultPromise = spawnAsync('eslint-config-prettier-check', [], { env });
-  const { child } = resultPromise;
-  child.stdin.setEncoding('utf8');
-  child.stdin.write(configString);
-  child.stdin.end();
+  const resultPromise = spawnAsync('eslint-config-prettier', [configFile], { env });
```

Comment:
This doesn't test `configFile` -- it tests the linter config that would have been applied to `configFile` is correct, which does two undesirable things: it loads `.eslintrc.js` and it computes the config for `configFile` specifically instead of checking the generic config.

One way to test this is to put `rules: { 'no-mixed-operators': 'error' }` inside of `web.js`. The tests should fail. 

Similarly, if `'no-mixed-operators': 'error'` is added to `node.js`, only the node test should fail and the other configurations should succeed.

I am not convinced this is possible with the new `eslint-config-prettier` script and may require more work.

Reply:
Thank you for the hint, I will try to reproduce recommended fails locally and will be back with results. 🙂

Reply:
If I put `'no-mixed-operators': 'error'` in `node.js` all 4 tests are failing:
<img width="794" alt="Screenshot 2021-08-03 at 12 50 29" src="https://user-images.githubusercontent.com/719641/128003675-bc34c569-2af3-47a3-94da-d379297691c0.png">

If I put `rules: { 'no-mixed-operators': 'error' }` in `web.js` all tests are successful.

Reply:
The tests were checking that the configs don't include rules that conflict with the ESLint Prettier config's rules.

The result you see above is what we want to avoid. The desired behavior is:
* When you add `'no-mixed-operators': 'error'` to node.js, only the `node.js` test should fail and the rest should succeed.
* When you add `'no-mixed-operators': 'error'` to web.js, only the `web.js` test should fail and the rest should succeed.

Reply:
I can understand what should happen, but I'm not understanding why the tests fails in way they does.

I have looked at several resources and the refactor I have made looks like the correct way to update previous `eslint-config-prettier-check` setup:
* https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md#version-700-2020-12-05

However, I can see the same results by running `eslint-config-prettier` directly from terminal on the config files (change to one file, results in error when checking all configs):
```sh
eslint-config-prettier ./node.js
# or even
npx eslint-config-prettier ./node.js
```

According to the changelog, instead of using stout from `eslint --print-config` they are using now new ESLint API to resolve config file, so maybe the way in which ESLint resolves config parts changed or there is a bug/quirk within this API which results in this behaviour. But still that are just the blind shots, because unfortunately I was not able to crack this problem.

Reply:
The intent of the old test was to validate each config.

As I understand it, `eslint-config-prettier <file>` now validates the config for `<file>`. We do not want to validate the linter config for `node.js` (that is, the config specified in .eslintrc.js, which is unrelated to this test). We want to validate the linter config exported from `node.js`.

I am not convinced that there is an obvious path forward using the scripts that eslint-config-prettier provides. It may require careful thinking to implement the desired behavior sustainably.

Reply:
The migration steps to which you linked aren't directly applicable. In the old test code, the key is that we pass in `'--config', configFile`.